js/index.js

```diff
@@ -1,42 +1,125 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
 
-  //... your code goes here
+
+
+// 1. find the price and quantity value of the product
+  let price = product.querySelector('.price span').innerText;
+  let quantity = product.querySelector('.quantity input').value;
+
+let subtotalNumber = price * quantity
+// console.log(subtotalNumber)
+
+//2.  manipulate/update DOM with  new value 
+product.querySelector('.subtotal span').innerHTML  = subtotalNumber;
+
+return subtotalNumber
+
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
+  
+  // get total products 
+  let products = document.getElementsByClassName("product");
+  
+  let prodctsPrice = updateSubtotal(products[0])
 
-  // ITERATION 2
-  //... your code goes here
+  let Sum = 0;
+  totalSum = document.getElementById('total-value')
+ 
+  let allprices = document.querySelectorAll('.subtotal span').value
+ 
+  for (i = 0; i < products.length; i++) {
+
+    updateSubtotal(products[i]) 
+    Sum += updateSubtotal(products[i])
+  }
+
+  totalSum.innerHTML = "Total: " + "$" +Sum 
 
-  // ITERATION 3
-  //... your code goes here
 }
 
-// ITERATION 4
 
+
+// ITERATION 4
 function removeProduct(event) {
+  
   const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
+  
+  target.parentNode.parentNode.remove();
+
+
 }
 
 // ITERATION 5
 
+
+// let productInputText = productInput.querySelector("text");
+
+// let productInputNumber = productInput.getElementsByTagName("input [2]");
+
+
+
 function createProduct() {
   //... your code goes here
+
+
+  let productInput = document.querySelectorAll(".create-product input")[0].value;
+  let productInput2 = document.querySelectorAll(".create-product input")[1].value;
+
+  
+let NewProduct = document.createElement(`tr`)
+NewProduct.innerHTML = `
+<td class="name">
+  <span></span>
+</td>
+<td class="price">$<span></span></td>
+<td class="quantity">
+  <input type="number" value="0" min="0" placeholder="Quantity" />
+</td>
+<td class="subtotal">$<span>0</span></td>
+<td class="action">
+  <button class="btn btn-remove">Remove</button>
+</td>`
+
+
+let nw1 = NewProduct.querySelectorAll('span')[0]
```

Comment:
Good idea! You could also use interpolation to insert the values into the string (`${variable1}`)